src/VscPreviewExtension/Microsoft.DocAsCode.Dfm.VscPreview/DocfxProcessor.cs

```diff
@@ -13,69 +14,133 @@ namespace Microsoft.DocAsCode.Dfm.VscPreview
+        }
+
+        private static string DocfxProcessCore(string baseDir, string relativePath, string markdownContent,
+            bool isFirstTime = false, string previewFilePath = null, string pageUpdateJsFilePath = null)
+        {
+            if (string.IsNullOrEmpty(baseDir))
+            {
+                throw new DocfxPreviewException("Base directory error");
+            }
+            if (string.IsNullOrEmpty(relativePath))
+            {
+                throw new DocfxPreviewException("Relative path error");
+            }
+            var markupResult = DfmMarkup(baseDir, relativePath, markdownContent);
+            if (!isFirstTime)
+            {
+                return markupResult;
+            }
+
+            if (string.IsNullOrEmpty(previewFilePath))
+            {
+                throw new DocfxPreviewException("Preview file path error");
+            }
+            if (string.IsNullOrEmpty(pageUpdateJsFilePath))
+            {
+                throw new DocfxPreviewException("Page Update js file path error");
+            }
 
-            var markupResult = DfmMarkup(baseDir, relativePath, markdownContent.ToString());
+            PreviewJsonConfig config = PreviewCommand.ParsePreviewCommand(baseDir);
 
-            string originHtmlPath = FindOriginHtml(baseDir, relativePath, config.OutputFolder);
+            // Path of Html which generated from target markdown file
+            string targetHtmlPath = Path.Combine(Path.GetDirectoryName(previewFilePath),
+                Path.GetFileNameWithoutExtension(relativePath) + ".html");
 
-            if (string.IsNullOrEmpty(originHtmlPath))
+            if (string.IsNullOrEmpty(targetHtmlPath))
             {
                 // TODO: If the return value is not a complete Html, it should be contacted with an Html header and tail
                 return markupResult;
             }
 
-            string htmlString = File.ReadAllText(originHtmlPath);
+            string htmlString = File.ReadAllText(targetHtmlPath);
 
             CQ dom = htmlString;
 
-            // Update markup result
-            dom.Select(config.MarkupResultLocation).Html(markupResult);
+            CQ addElements = $"<script type='text/javascript' src='{pageUpdateJsFilePath}'></script>" +
+                             $"<meta name='pageRefreshFunctionName' content ='{config.PageRefreshFunctionName}'>" +
+                             $"<meta name='port' content='{config.Port}'>" +
+                             $"<meta name='filePath' content='{relativePath}'>" +
+                             $"<meta name='markupTagType' content='{config.MarkupTagType}'>" +
+                             $"<meta name='markupClassName' content='{config.MarkupClassName}'>";
+
+            // Append page update js
+            dom.Select("script")
+                .Last()
+                .After(
+                    addElements.Elements.First());
+
+            for (int i = 1; i < addElements.Length; i++)
```

Comment:
cause the first one is a script node and it have been added before this.But we can merge this

---
In reply to: [105340084](https://github.com/dotnet/docfx/pull/1397#discussion_r105340084) [](ancestors = 105340084)